libgstc/libgstc_socket.c

```diff
@@ -49,17 +49,41 @@ struct _GstcSocket
 {
   int socket;
   struct sockaddr_in server;
+  int keep_connection_open;
 };
 
+int
```

Comment:
if these methods are private, they should be static. And we should have an independent declaration.